src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -2215,9 +2229,10 @@ int wm_vuldet_insert(wm_vuldet_db *parsed_oval, update_node *update) {
                 return wm_vuldet_sql_error(db, stmt);
             }
             wdb_finalize(stmt);
-            free(info_it->refs->values[j]);
         }
 
+        free_strarray(info_it->refs->values);
```

Comment:
We were only freeing the memory of each value in the array but not the array itself.